pkgs/build-support/rust/hooks/default.nix

```diff
@@ -68,15 +68,37 @@ in {
         # The `.nativeDrv` stanza works like nativeBuildInputs and ensures cross-compiling has the right version available.
         diff = "${diffutils.nativeDrv or diffutils}/bin/diff";
 
-        # Target platform
-        rustTarget = ''
-          [target."${rust.toRustTarget stdenv.buildPlatform}"]
+        # We want to specify the correct crt-static flag for both
+        # the build and host platforms. This is important when the wanted
+        # value for crt-static does not match the defaults in the rustc target,
+        # like for pkgsMusl or pkgsCross.musl64; Upstream rustc still assumes
+        # that musl = static[1].
+        #
+        # By default, Cargo doesn't apply RUSTFLAGS when building build.rs
+        # if --target is passed, so the only good way to set crt-static for
+        # build.rs files is to use the unstable -Zhost-config Cargo feature.
+        # This allows us to specify flags that should be passed to rustc
+        # when building for the build platform. We also need to use
+        # -Ztarget-applies-to-host, because using -Zhost-config requires it.
+        #
+        # When doing this, we also have to specify the linker, or cargo
+        # won't pass a -C linker= argument to rustc.  This will make rustc
+        # try to use its default value of "cc", which won't be available
+        # when cross-compiling.
+        #
+        # [1]: https://github.com/rust-lang/compiler-team/issues/422
+        cargoConfig = ''
+          [host]
           "linker" = "${ccForBuild}"
```

Comment:
This is a bit of a necropost, but this looks like a bug: `ccForHost` is used for the target, and `ccForBuild` is used for the host side.

Reply:
It's not.  What Rust calls "host" is what we call "build", and what Rust calls "target" is what we call "host".